common/src/main/java/com/revenuecat/purchases/common/offlineentitlements/ProductEntitlementMappings.kt

```diff
@@ -31,4 +36,14 @@ data class ProductEntitlementMappings(
     fun toMap(): Map<String, List<String>> {
         return mappings.associateBy({ it.identifier }, { it.entitlements })
     }
+
+    fun toJson() = JSONObject().apply {
+        val mappingsArray = mappings.map {
+            JSONObject().apply {
+                put(PRODUCT_ID_KEY, it.identifier)
+                put(ENTITLEMENTS_KEY, JSONArray(it.entitlements))
+            }
+        }
+        put(PRODUCTS_KEY, JSONArray(mappingsArray))
+    }
```

Comment:
I thought about just storing the original rawJson instead of mapping it back to the original Json like this, but I wanted to decouple the data object from the network layer